### PR TITLE
Component RBAC for different roles on user token

### PR DESCRIPTION
**Pull request by urandu**
In this change, we are introducing RBAC for components. This is because there are some components that you only want certain roles to access. 
Example usage of this component is as highlighted below:

```
     <ComponentRBAC allowedRoles={['admin', 'member']}>
        <CCard className="mb-4">
        </CCard>
      </ComponentRBAC>
```

### CODE REVIEW

File: src/views/dashboard/Dashboard.js [hunk 1]
```diff
@@ -177,133 +178,134 @@ const Dashboard = () => {
 
   return (
     <>
-      <CCard className="mb-4">
```
Comment (lakivisi) [inline src/views/dashboard/Dashboard.js:180]: Can we move this out to a template file for easy refactoring later?

Comment (urandu) [reply src/views/dashboard/Dashboard.js:180]: This is actually just place holder, it should be deleted.

Comment (urandu) [reply src/views/dashboard/Dashboard.js:180]: @lakivisi The actual dashboard will come at a later point. In this change we are mainly introducing the `<ComponentRBAC />` component and logic.

**Review by leroy-netizen — APPROVED** [review]

**Review by leroy-netizen — APPROVED** [review]

File: src/views/dashboard/Dashboard.js (hunk 1 above)
Comment (urandu) [reply src/views/dashboard/Dashboard.js:180]: Yes, the component will nest other components that we need to control access to.

**Pull request merged**